realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -196,11 +193,6 @@ public void run() {
     // doCreateRealmOrGetFromCache is called with globalCount == 0.
     private RealmConfiguration configuration;
 
-    // Column indices are cached to speed up opening typed Realm. If a Realm instance is created in one thread, creating
-    // Realm instances in other threads doesn't have to initialize the column indices again.
-    private static final int MAX_ENTRIES_IN_TYPED_COLUMN_INDICES_ARRAY = 4;
-    private final ColumnIndices[] typedColumnIndicesArray = new ColumnIndices[MAX_ENTRIES_IN_TYPED_COLUMN_INDICES_ARRAY];
```

Comment:
Did you measure how long it takes to generate the column indices pr. thread? I would assume it isn't much, but having some idea would be nice. Given how much simpler the code is gets, it is probably fine either way.